### PR TITLE
Verified 1:1 chats, Android UI

### PR DESCRIPTION
**Pull request by Hocuri**
Depends on https://github.com/deltachat/deltachat-core-rust/pull/4315/.

Follow-up for https://github.com/deltachat/deltachat-android/pull/2541.

- Show a "verified" icon on protection changed update messages
- Add C API
- Show a dialog over the input bar not only for contact requests, but also when the protection is broken (using the new method `isHalfBlocked()`)
  - The positive button reads "OK" (as opposed to contact requests where it reads "Accept")
- Add SVG icons, remove PNG one
- Translations
- When tapping on the `DC_INFO_PROTECTION_{EN|DIS}ABLED` message, show more information (for now, it leads to the online preview of my FAQ PR)
- Block loading remote images in the "Full Msg View" not only for contact requests, also when the protection is broken (using the new method `isHalfBlocked()`)
- Show a big verified/crossed-out-verified symbol over the `DC_INFO_PROTECTION_{EN|DIS}ABLED` messages
- Fix a bug that was kind of present before (for contact requests) but only became really visible now:
  - Set a draft with an image (or other attachment)
  - Your chat partner breaks verification
  - Expected: Both the input bar and the draft image are hidden by the input-bar-dialog.
  - Bug behavior (before c31de5bcf): The input bar is hidden, but the draft image stays visible.

TODO before merging:
- [x] Clean up, settle for a core API
- [x] Translations
- [x] Make taps work
- [x] Change back the verified icon (we need an SVG)
- [x] Revert updating submodule (I updated it for the preview APK to work)

cc @r10s

### CODE REVIEW

File: src/org/thoughtcrime/securesms/connect/DcHelper.java
Comment (r10s) [reply src/org/thoughtcrime/securesms/connect/DcHelper.java]: probably okayish in most languages, we do not add full stop or so here, it could also be two views. in case issues pop up, we can add the two additional strings.

Comment (r10s) [reply src/org/thoughtcrime/securesms/connect/DcHelper.java]: otoh, thinking it over, the "hack" would be required on all, so, maybe better add the two additinal strings and keep the setTranslation() code clean and straight-forward

File: src/com/b44t/messenger/DcChat.java
Comment (adbenitez) [inline src/com/b44t/messenger/DcChat.java:67]: "half blocked" I love this :)

File: jni/deltachat-core-rust
Comment (Hocuri) [inline jni/deltachat-core-rust:1]: Before merging, I need to remember to revert updating the submodule.

**Review by adbenitez — APPROVED** [review]
nice, as discussed IRL, I had some small opinions etc. but I think this is a fine solution as well and a great improvement! :rocket:

File: res/values/strings.xml
Comment (Hocuri) [reply res/values/strings.xml:415]: We now changed the string to "OK", so, this message is wrong now

Comment (Hocuri) [reply res/values/strings.xml]: I think it's OK to not translate it for now: e27ac97e0

**Review by adbenitez — APPROVED** [review]
that the "this chat is now e2e encrypted" got added at the end of the chat for long-ago verified contacts and apparently changing the sorting order of the chat as if they were verified recently felt a bit weird to me, also the banner appeared after the actual message, which made it look like the message before that was not safe, when in fact the verification happened long ago before the recent message from today, I sent you some pictures via delta chat

**Comment by Hocuri** [comment]
Nice findings! They are all to be fixed in core, so, nothing blocking merging this. The main question at this point is whether it's a problem for the release.

File: res/values/strings.xml
Comment (r10s) [inline res/values/strings.xml:834]:
```suggestion
    <string name="chat_protection_broken_explanation">End-to-end encryption cannot be guaranteed anymore, likely because %1$s reinstalled Delta Chat or sent a message from another device.\n\nYou may meet them in person and scan their QR Code again to reestablish guaranteed end-to-end encryption.</string>
```

we most times add a space between "end-to-end" and the next word

**Review by r10s — APPROVED** [review]
great pr! code-wise, things look good to me.

by the large changes, also in core, i wouldn't be surprised about some hickup, so we should select the moment of merging and releasing carefully (as always :) so that there are ppl how can take care about hickup on releasing :)

**Comment by Hocuri** [comment]
As agreed in the "delta android" chat, I created a `stable` branch that doesn't contain this PR and then merged.

**Comment by Hocuri** [comment]
**What UIs need to do to adapt to Verified 1:1 chats:**

- Make sure that 1:1 chats show a verified icon exactly if `chat.is_protected()` returns `true` (this has been done in https://github.com/deltachat/deltachat-android/pull/2541 for Android - the correct behavior didn't change here, this was just a bug in Android).
  - We didn't discuss yet when exactly a verified icon should be shown in the contact's profile.
- When tapping on the `DC_INFO_PROTECTION_{EN|DIS}ABLED` message, show more information:
  - `DC_INFO_PROTECTION_ENABLED` directly opens https://staging.delta.chat/684/en/help#whatdoesverifiedmean
  - `DC_INFO_PROTECTION_DISABLED` opens a dialog that shows `chat_protection_broken_explanation` as an explanation. This dialog has buttons "Verify", OK", and "Learn More". "Learn More" leads to https://staging.delta.chat/684/en/help#verificationbroken".
  - The links are going to change when the FAQ update is ready. Note that they lead to `staging.delta.chat` for now.
- Show a "verified" icon over `DC_INFO_PROTECTION_{EN|DIS}ABLED` messages
  - There are SVG icons: [enabled](https://github.com/deltachat/interface/blob/master/icons/ic_verified.svg), [disabled](https://github.com/deltachat/interface/blob/master/icons/ic_verified_broken.svg)
    - If you were using a PNG "verified" icon, you can remove it and use the SVG icon instead
- Add C API `dc_chat_is_protection_broken()` if necessary
- Show a dialog over the input bar when the protection is broken. You can re-use the same dialog you're using for contact requests; DC Android defines a method [`isHalfBlocked()`](https://github.com/deltachat/deltachat-android/pull/2560/files#diff-cb9cdc9a8ff1759df4ce3216a62dc153fa6b008a091bbec23eb66452dba5031dR67-R69) for this.
  - The positive button reads "OK" (as opposed to contact requests where it reads "Accept")
- There are two new stock translations which need to be set in Core using `set_stock_translation()`
- Block loading remote images in the "Full Msg View" not only for contact requests, also when the protection is broken (using the new method `isHalfBlocked()`)

**Pull request merged**
